spec/execution_tests.lua

```diff
@@ -1749,6 +1749,42 @@ function execution_tests.run(compile_file, backend, _ENV, only_compile)
         end)
     end)
 
+    describe("For-in loops", function()
```

Comment:
Currently these tests cases are testing if the code is compiling successfully but are not testing that they produce the expected result when you run the code. We need to add an `it` block to do that.

Furthermore, I think it would be nice if we had a more intuitive example. One possibility would be to create a custom iterator function called `rev_ipairs` that works similarly to ipairs, except that it iterates in decreasing order instead of increasing order.

BTW, while I like the idea of testing loops of the form `for i, x in a, b, c`, I think we should also test the more common pattern where the RHS is a function call like `for i, x in rev_ipairs(xs)`.